src/ToDo/Presentation/WelcomeViewModel.cs

```diff
@@ -4,26 +4,30 @@
 public partial class WelcomeViewModel
 {
 	private readonly IAuthenticationService _authService;
+	private readonly IUserProfilePictureService _userSvc;
 	private readonly INavigator _navigator;
 	private readonly IDispatcher _dispatcher;
 
 	private WelcomeViewModel(
 		IDispatcher dispatcher,
 		INavigator navigator,
+		IUserProfilePictureService userSvc,
 		IAuthenticationService authService)
 	{
 		_dispatcher = dispatcher;
 		_navigator =navigator;
 		_authService = authService;
+		_userSvc = userSvc;
 	}
 
 	public ICommand GetStarted => Command.Async(DoGetStarted);
 	private async ValueTask DoGetStarted(CancellationToken ct)
 	{
 		var user = await _authService.AuthenticateAsync(_dispatcher);
-
 		if(user is not null)
 		{
+			var profilePicture = await _userSvc.GetAsync(cancellationToken: ct);
+			_authService.SetUserProfilePicture(profilePicture);
 			await _navigator.NavigateRouteAsync(this, string.Empty, cancellation: ct);
 		}
```

Comment:
Please create a helper method in the AuthService to do that. The setting of the profile picture data should not be done in a ViewModel.

Reply:
I upload the changes, can you check again, please?